website/guide/programming.html

```diff
@@ -493,6 +493,32 @@ <h2 id="duktape-api">Duktape API</h2>
 duk_get_prop_string(ctx, -1, "my_state_variable");
 </pre>
 
+
+<h3>Hidden Symbol properties</h3>
+
+<p>If data needs to be associated with an object, but hidden from Ecmascript
+code, hidden <a href="#symbols">Symbols</a> can be used as a property key. The
+key is only accessible via the C API, unless passed to Ecmascript code from the
+C API. A common use case is to associated backing pointers/data to C memory.
+Symbols are created as a string, but with a specific prefix.</p>
+
+<p>Setting then getting a hidden symbol on <code>this</code>.</p>
+<pre class="c-code">
+duk_push_this(ctx);
+duk_push_pointer(ctx, my_context_data);
+duk_put_prop_string(ctx, -2, "\xff" "\xff" "my_context_data");
+/* ... */
+duk_push_this(ctx);
+duk_get_prop_string(ctx, -1, "\xff" "\xff" "my_context_data");
+my_context_data_t *my_context_data = duk_get_pointer(ctx, -1);
+</pre>
+
+<p>See the <a href="#symbols">Symbols</a> section regarding the strange C
+literal string syntax.</p>
+
+<p>Add a finalizer function to the Ecmascript object if you need to clean up
+resources.</p>
```

Comment:
This remark seems disconnected to the topic. Finalizers can be used also for state which is stored as Ecmascript visible strings or Symbols. I'd suggest removing, or rewording to better connect it to the discussion above (for example: "If the data stored using hidden Symbols needs native code cleanup when the object is freed, a finalizer function ...", and make the finalizer a link to the finalization section).

Reply:
I think the rewording/linking would help make the section more straight forward and help to keep copy-pasters in check?

Reply:
After looking over the state section more, I think moving to a more general note in the state section as a whole would be better as it can apply to all of the state techniques. I don't see an immediate place to put it, so will leave it out of this PR and think about a more cohesive note later.